File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mins=prices[0];    
        int maxs=0;
        for(int i=0;i<prices.length;i++){
            int t=prices[i]-mins;
            mins=Math.min(mins,prices[i]);
            maxs=Math.max(maxs,t);
        }
        return maxs;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int minprofit=0;
        int maxprofit=prices[0];
        for(int i=0;i<prices.size();i++){
            int h=prices[i]-maxprofit;
            cout<<h;
            maxprofit=min(maxprofit,prices[i]);
            minprofit=max(minprofit,h);
        }
        return minprofit;
    }
};